cfme/containers/image.py

```diff
@@ -116,7 +114,33 @@ def compliant(self):
             raise ValueError("{} is not a known state for compliance".format(text))
 
 
+@attr.s
+class ImageCollection(BaseCollection):
+    """Collection object for :py:class:`Image`."""
+
+    ENTITY = Image
+
+    def all(self):
+
+        image_table = self.appliance.db.client['container_images']
+        ems_table = self.appliance.db.client['ext_management_systems']
+        image_query = self.appliance.db.client.session.query(
+            image_table.name, image_table.image_ref, ems_table.name).join(
+            ems_table, image_table.ems_id == ems_table.id)
+        images = []
+
+        for name, id, provider_name in image_query.all():
+            images.append(self.instantiate(name=name, id=id,
+                provider=ContainersProvider(name=provider_name, appliance=self.appliance)))
+        return images
+
+    def get_random_instances(self, count=1):
+
+        return random.sample(self.all(), count)
+
+
 @navigator.register(Image, 'All')
```

Comment:
Remove this. It should only be registered to the ImageCollection

Reply:
We spook with Pete, he say that it is OK for now, so we leave it as is.

Reply:
**Required**

I'd like to revisit the allowance of this during the transition to collections - I think that we need to enforce registering the 'All' destination against the collection only. It is trivial to navigate against the entity's parent in order to get to the 'All' destination.

Reply:
I remove this line according to your suggestion,
Now it registering the 'All' destination against the collection only.